### DISCUSSION TITLE
Layer & 'Visible image' sampling options

### DISCUSSION THREAD

**Discussion by ichordev** [Orama-Interactive/Pixelorama#622] [Ideas]
The fill and the colour picker tools could benefit tremendously from an option to switch from layer sampling to image sampling.

For instance, the colour picker normally selects colour from the selected layer. Selecting image sampling means that it would select from what you see in your canvas. This is often a more intuitive behaviour.

For the fill tool, this is much the same. With image sampling selected filling would still only affect the current layer, but would check every visible layer to determine where to fill, instead of only the selected one.

Since the canvas already displays a composite of every visible layer merged together, I hope this feature wouldn't be too tricky to implement.

**Comment by narcatix** [comment]
I agree! That would be amazing!
